### PR TITLE
[Fix][Connector-V2]Oceanbase vector database is added as the source server

### LINKED ISSUE #7831: [Fix][Connector-V2]Oceanbase vector database is added as the source server
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Currently, the ResultSetMetaData of oceanbase does not support obtaining vector information, but you can obtain it through DatabaseMetaData

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xxsc0529**
### Purpose of this pull request
Currently, the ResultSetMetaData of oceanbase does not support obtaining vector information, but you can obtain it through DatabaseMetaData.This method is used to parse vector types with OceanBase as the source. close #7831

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java Added OceanBase vector library to milvus testing

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java [hunk 1]
```diff
@@ -201,8 +209,7 @@ public CatalogTable getTable(String sqlQuery) throws SQLException {
         try (Statement statement = defaultConnection.createStatement();
                 ResultSet resultSet = statement.executeQuery(sqlQuery)) {
             ResultSetMetaData metaData = resultSet.getMetaData();
-            return CatalogUtils.getCatalogTable(
-                    metaData, new OceanBaseMySqlTypeMapper(typeConverter), sqlQuery);
+            return getCatalogTable(metaData, defaultConnection.getMetaData(), sqlQuery);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: shall we update https://github.com/apache/seatunnel/blob/e4e12245b90e994bce27ea90ac044e722888a8ae/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java#L130 too?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMySqlTypeConverter.java [hunk 2]
```diff
@@ -510,4 +554,142 @@ public BasicTypeDefine<OceanBaseMysqlType> reconvert(Column column) {
 
         return builder.build();
     }
+
+    public static List<Column> convert(DatabaseMetaData metadata, TablePath tablePath)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMySqlTypeConverter.java]: Can we merge this into https://github.com/apache/seatunnel/blob/e4e12245b90e994bce27ea90ac044e722888a8ae/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMySqlTypeConverter.java#L109 ?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java (hunk 1 above)
Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: No, I reused it

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMySqlTypeConverter.java (hunk 2 above)
Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMySqlTypeConverter.java]: Yes, I've incorporated this method

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java [hunk 3]
```diff
@@ -217,4 +218,49 @@ public String getExistDataSql(TablePath tablePath) {
                 "SELECT * FROM `%s`.`%s` LIMIT 1;",
                 tablePath.getDatabaseName(), tablePath.getTableName());
     }
+
+    public CatalogTable getCatalogTable(Connection defaultConnection, String sqlQuery)
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: how about put this part into `public CatalogTable getTable(String sqlQuery) throws SQLException {`?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java [hunk 4]
```diff
@@ -217,4 +218,49 @@ public String getExistDataSql(TablePath tablePath) {
...
+        }
+        databaseName = StringUtils.isBlank(databaseName) ? null : databaseName;
+        schemaName = StringUtils.isBlank(schemaName) ? null : schemaName;
+        TablePath tablePath =
+                StringUtils.isBlank(tableName)
+                        ? TablePath.DEFAULT
+                        : TablePath.of(databaseName, schemaName, tableName);
+        PreparedStatement ps = defaultConnection.prepareStatement(getSelectColumnsSql(tablePath));
+        ResultSet resultSet = ps.executeQuery();
+        ResultSet primaryKeys =
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: please add try resource, we should close it after used.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java [hunk 5]
```diff
@@ -217,4 +218,49 @@ public String getExistDataSql(TablePath tablePath) {
...
+    public CatalogTable getCatalogTable(Connection defaultConnection, String sqlQuery)
+            throws SQLException {
+        TableSchema.Builder schemaBuilder = TableSchema.builder();
+        String tableName = null;
+        String databaseName = null;
+        String schemaName = null;
+        String catalogName = "jdbc_catalog";
+        try {
+            Statement statement = defaultConnection.createStatement();
+            ResultSet resultSet = statement.executeQuery(sqlQuery);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: ditto

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java (hunk 3 above)
Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: ok

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java [hunk 6]
```diff
@@ -281,6 +286,63 @@ public void testFakeToOceanBase(TestContainer container)
         }
     }
 
+    @TestTemplate
+    public void testOceanBaseToMilvus(TestContainer container) throws Exception {
+        try {
+            initOceanBaseTestData();
+            Container.ExecResult execResult = container.executeJob(configFile().get(2));
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]:
```suggestion
            Container.ExecResult execResult = container.executeJob("/jdbc_oceanbase_source_and_milvus_sink.conf");
```
This way is more stable

**Review by Hisoka-X — COMMENTED** [review]
Please update other test case too. https://github.com/apache/seatunnel/blob/69086e0dba1195822658f7618ac4c53d26d9ba45/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java#L266

**Review by Hisoka-X — DISMISSED** [review]
LGTM. Thanks @xxsc0529

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**